framework/src/mysql/mysql.js

```diff
@@ -341,10 +341,16 @@ const getTableInstance = async (...tableParams) => {
 			params.search = Array.isArray(params.search) ? params.search : [params.search];
 
 			params.search.forEach(search => {
-				const { property, pattern, startsWith, endsWith } = search;
-				if (pattern) query.where(`${property}`, 'like', `%${pattern}%`);
-				if (startsWith) query.where(`${property}`, 'like', `${startsWith}%`);
-				if (endsWith) query.where(`${property}`, 'like', `%${endsWith}`);
+				const { property, pattern, startsWith, endsWith, allowWildCards } = search;
+				if (allowWildCards === true) {
+					if (pattern) query.where(`${property}`, 'like', `%${pattern}%`);
+					if (startsWith) query.where(`${property}`, 'like', `${startsWith}%`);
+					if (endsWith) query.where(`${property}`, 'like', `%${endsWith}`);
+				} else {
+					if (pattern) query.where(`${property}`, 'like', `%${pattern.replace(/%/g, '\\%').replace(/_/g, '\\_')}%`);
+					if (startsWith) query.where(`${property}`, 'like', `${startsWith.replace(/%/g, '\\%').replace(/_/g, '\\_')}%`);
+					if (endsWith) query.where(`${property}`, 'like', `%${endsWith.replace(/%/g, '\\%').replace(/_/g, '\\_')}`);
```

Comment:
Please use `replaceAll` instead of `replace`. Kindly, make the relevant changes elsewhere.
To cover the same, let's new testcases. We can discuss them offline.